### PR TITLE
[JENKINS-41631] Running requireUpperBoundDeps

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-41631](https://issues.jenkins-ci.org/browse/JENKINS-41631)

Downstream of https://github.com/apache/maven-enforcer/pull/23 and https://github.com/jenkinsci/jenkins-test-harness/pull/62.

@reviewbybees

### CODE REVIEW

**Comment by jglick** [comment]
https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/38 failed against 2.47 due to:

```
Require upper bound dependencies error for org.kohsuke:access-modifier-annotation:1.4 paths to dependency are:
+-org.jenkins-ci.plugins.workflow:workflow-cps-global-lib:2.9-SNAPSHOT
  +-org.jenkins-ci.main:jenkins-core:2.47
    +-org.kohsuke:access-modifier-annotation:1.4
and
+-org.jenkins-ci.plugins.workflow:workflow-cps-global-lib:2.9-SNAPSHOT
  +-org.jenkins-ci.main:jenkins-core:2.47
    +-org.jenkins-ci.main:cli:2.47
      +-org.kohsuke:access-modifier-annotation:1.7
```

This is fixed by https://github.com/jenkinsci/jenkins/pull/2866 in 2.61 but perhaps `org.kohsuke:access-modifier-annotation` needs to be excluded so as to permit building against intermediate versions, especially 2.60.x which will not get this backport.

**Comment by jglick** [comment]
Inconsistency introduced by https://github.com/jenkinsci/jenkins/pull/2656 in 2.36.

**Comment by jglick** [comment]
@reviewbybees done

On holding pending MENFORCER-273 release.

**Comment by jglick** [comment]
[call for release](http://mail-archives.apache.org/mod_mbox/maven-dev/201706.mbox/%3CCA%2BnPnMxOzo6HJZ44Gp9Oc6kGr1xxWc0HwJVyHwvdCGOxgio-cQ%40mail.gmail.com%3E) by @stephenc today

**Comment by jglick** [comment]
1.4.2 released but [not yet on Central](http://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22org.apache.maven.plugins%22%20AND%20a%3A%22maven-enforcer-plugin%22).

**Comment by jglick** [comment]
Seems @stephenc had been planning to release this but ran into some technical issues. I may just push a temporary `1.4.1.jenkins-1` to our internal repo to unblock the POM release if there are no objections.

**Comment by stephenc** [comment]
@jglick go with pushing to the jenkins repo with -jenkins-1

**Comment by oleg-nenashev** [comment]
I have also created https://github.com/apache/maven-enforcer/pull/25/ to address the case with test dependencies

File: pom.xml [hunk 1]
```diff
@@ -321,7 +321,7 @@
         </plugin>
         <plugin>
           <artifactId>maven-enforcer-plugin</artifactId>
-          <version>1.4.1</version>
+          <version>1.4.2.jenkins-1</version> <!-- TODO update to official 1.4.2 or 3.0.0 when available -->
```
Comment (jglick) [inline pom.xml]: 3.0.0-M1 is also available.

**Pull request merged**